Agora manufacturing data pipeline README (#304)
Azure/arc_jumpstart_docs: docs/azure_jumpstart_ag/manufacturing/contoso_motors/data_opc/_index.md

---
type: docs
weight: 2
title: Data pipeline and reporting across cloud and edge for Contoso Motors
linkTitle: Data pipeline and reporting across cloud and edge for Contoso Motors
summary: |
  WIP
serviceOrPlatform: Manufacturing
technologyStack:
  - AZURE DATA EXPLORER
  - AZURE EVENT HUB
  - AZURE EVENT GRID
---

# Data pipeline and reporting across cloud and edge for Contoso Motors

## Overview

One of Contoso's biggest objectives is how to use the data coming from their manufacturing assembly line, welding robots, battery assemblies and visualize it for business intelligence by leveraging the power of the cloud.

> In this scenario, Contoso wants to use their data pipeline so that data coming from cars manufactured at various plants flow to [Azure Data Explorer (ADX)](https://learn.microsoft.com/azure/data-explorer/data-explorer-overview) database and generate near real-time reports for production and operational efficiency. By doing so, Contoso management can leverage these reports to adjust their production, improve quality, reduce wastage, and save manufacturing cost.

Reviewer: ```suggestion
In this scenario, Contoso wants to use their data pipeline so that data coming from automobile parts manufactured at various plants flow to [Azure Data Explorer (ADX)](https://learn.microsoft.com/azure/data-explorer/data-explorer-overview) database and generate near real-time reports for production and operational efficiency. By doing so, Contoso management can leverage these reports to adjust their production, improve quality, reduce wastage, and save manufacturing cost.
```

![Screenshot showing the data pipeline architecture diagram](./img/contoso_motors-datapipeline_architecture.png)

## Contoso Motors dashboard reports

Contoso supports dashboard reports for the manufacturing analytics and monitoring. These reports are created in ADX to allow users to view dashboards reports. These reports are generated based on live data received from the MQTT Broker into the ADX database using data integration.

## Manually import dashboards

> > **Note:** If you used the [Azure Developer CLI (azd) method](../deployment/#deployment-via-azure-developer-cli) to deploy the Contoso Motors scenario, you may skip this section as these reports are automatically imported for you during the automated deployment.

Reviewer: ```suggestion
```

We decided to remove azd from the readmes

  ![Screenshot showing the dashboard report JSON file to import](./img/adx_select_dashboard_file.png)

- Confirm the dashboard report name, accept the suggested name (or choose your own), and click Create.

  ![Screenshot showing the dashboard report name confirmation](./img/adx_confirm_dashboard_report_name.png)

- By default, there is no data available in the ADX manufacturing database to display in the report after deployment. Click Save to save the dashboard report in ADX.

  ![Screenshot showing the empty data in Contoso Motors dashboard report](./img/adx_manufacturing_report_empty_data.png)

  > **Note:** Depending on the type of user account being used to access ADX dashboards, you might have issues accessing data in the _manufacturing_ database in the ADX cluster with an error _User principal 'msauser=xyz@abc.com' is not authorized to read database 'manufacturing'_. If you experience this access issue, refer to [Jumpstart Agora - Contoso Motors scenario troubleshooting](../troubleshooting/#user-principal-is-not-authorized-to-read-database-manufacturing) guide to troubleshoot and address this access issue.

## Generate sample data using Data Emulator

> Once the Agora Contoso Motors scenarios is deployed and fully functions, MQTT simulators start sending assemblyline, battery, and welding data to Azure Event Hub and is ingested to ADX database using ADX database connection. While this data is realtime and takes time to produce enough data to view dashboards reports, this scenario comes with a Data Emulator tool available on the Agora client VM. Data Emulator can produce data for past several days or weeks to view dashboard reports. Use instructions below to generate sample data using the Data Emulator tool.

Reviewer: ```suggestion
Once the Agora Contoso Motors scenarios is deployed and fully functions, MQTT simulators start sending assemblyline, battery, and welding data to Azure Event Hub and is ingested to ADX database using ADX database connection. While this data is realtime, it takes time to produce enough data to view dashboards reports, this scenario comes with a Data Emulator tool available on the Agora client VM. Data Emulator can produce data for past several days or weeks to view dashboard reports. Use instructions below to generate sample data using the Data Emulator tool.
```

- On the Client VM, locate Data Emulator icon on the desktop.

  ![Screenshot showing the Data Emulator on the desktop](./img/locate_data_emulator_desktop.png)

- Double click on the Data Emulator desktop icon to launch executable and generate sample data. Confirm by entering option **1** for past data or **2** to start generating current data. Select 1 to generate past data.

  > **Note:** You can still generate additional sample data by running this tool multiple times, but there might be duplicate key errors and fails to generate data in subsequent attempts.

- Enter no of days to generate past data until current date and time.

  ![Screenshot showing the sample data generation confirmation](./img/confirm_sample_data_generation.png)

  ![Screenshot showing the generating sample data](./img/confirm_sample_data_past_days.png)

  ![Screenshot showing the generating sample data](./img/sample_data_generation.png)

> - Once the sample data generation is complete, from ADX Dashboards open Contoso Motors and Auto Parts dashboard report to view simulated manufacturing data. Allow some time to propagate data into the ADX database using an integrated data pipeline.

Reviewer: ```suggestion
- Once the sample data generation is complete, from [ADX Dashboards](https://dataexplorer.azure.com/) open Contoso Motors and Auto Parts dashboard report to view simulated manufacturing data. Allow some time to propagate data into the ADX database using an integrated data pipeline.
```
